src/Microsoft.Health.Fhir.Liquid.Converter/Models/PartialDateTime.cs

```diff
@@ -129,5 +129,39 @@ public string ToFhirString(TimeZoneHandlingMethod timeZoneHandling = TimeZoneHan
 
             return resultDateTime.ToString(dateTimeFormat);
         }
+
+        public string ToHl7v2Date(TimeZoneHandlingMethod timeZoneHandling = TimeZoneHandlingMethod.Preserve)
+        {
+            var resultDateTime = timeZoneHandling switch
+            {
+                TimeZoneHandlingMethod.Preserve => DateTimeValue,
+                TimeZoneHandlingMethod.Utc => DateTimeValue.ToUniversalTime(),
+                TimeZoneHandlingMethod.Local => DateTimeValue.ToLocalTime(),
+                _ => throw new ArgumentException(Resources.InvalidTimeZoneHandling),
+            };
+
+            if (Precision <= DateTimePrecision.Day)
+            {
+                return Precision switch
+                {
+                    DateTimePrecision.Day => resultDateTime.ToString("yyyyMMdd"),
+                    DateTimePrecision.Month => resultDateTime.ToString("yyyyMM"),
+                    DateTimePrecision.Year => resultDateTime.ToString("yyyy"),
+                    _ => throw new ArgumentException("Invalid dateTimeObject with empty Year field.")
+                };
+            }
+
+            string dateTimeFormat;
+            if (Precision < DateTimePrecision.Milliseconds)
+            {
+                dateTimeFormat = "yyyyMMddHHmmssK";
+            }
+            else
+            {
+                dateTimeFormat = MillisecondString == null ? "yyyyMMddHHmmssK" : $"yyyyMMddHHmmss{MillisecondString}K";
```

Comment:
nit: can combine ` || MillisecondString == null` with above check for readability, since they return the same value. Also, just in case this format was updated for one but missed for the other by mistake. But feel free to 'won't fix' if you feel this is more readable as is.

Reply:
thanks, updated